setupinfo.py

```diff
@@ -392,21 +390,73 @@ def check_min_version(version, min_version, error_name):
     return True
 
 
-def get_library_version(config_tool):
-    is_pkgconfig = "pkg-config" in config_tool
-    return run_command(config_tool,
-                       "--modversion" if is_pkgconfig else "--version")
+def get_library_version(prog, libname=None):
+    if libname:
+        return run_command(prog, '--modversion %s' % libname)
+    else:
+        return run_command(prog, '--version')
+
 
+PKG_CONFIG = None
+XML2_CONFIG = None
+XSLT_CONFIG = None
 
 def get_library_versions():
-    xml2_version = get_library_version(find_xml2_config())
-    xslt_version = get_library_version(find_xslt_config())
-    return xml2_version, xslt_version
+    global XML2_CONFIG, XSLT_CONFIG
+
+    # Pre-built libraries
+    if XML2_CONFIG and XSLT_CONFIG:
+        xml2_version = get_library_version(XML2_CONFIG)
+        xslt_version = get_library_version(XSLT_CONFIG)
+        return xml2_version, xslt_version
+
+    # Path to xml2-config and xslt-config specified on the command line
+    if OPTION_WITH_XML2_CONFIG:
+        xml2_version = get_library_version(OPTION_WITH_XML2_CONFIG)
+        if xml2_version and OPTION_WITH_XSLT_CONFIG:
+            xslt_version = get_library_version(OPTION_WITH_XSLT_CONFIG)
+            if xslt_version:
+                XML2_CONFIG = OPTION_WITH_XML2_CONFIG
+                XSLT_CONFIG = OPTION_WITH_XSLT_CONFIG
+                return xml2_version, xslt_version
+
+    # Try pkg-config
+    global PKG_CONFIG
+    PKG_CONFIG = os.getenv('PKG_CONFIG', 'pkg-config')
+    xml2_version = get_library_version(PKG_CONFIG, 'libxml-2.0')
+    if xml2_version:
+        xslt_version = get_library_version(PKG_CONFIG, 'libxslt')
+        if xml2_version and xslt_version:
+            return xml2_version, xslt_version
+
+    # Try xml2-config and xslt-config
+    XML2_CONFIG = os.getenv('XML2_CONFIG', 'xml2-config')
+    xml2_version = get_library_version(XML2_CONFIG)
+    if xml2_version:
+        XSLT_CONFIG = os.getenv('XSLT_CONFIG', 'xslt-config')
+        xslt_version = get_library_version(XSLT_CONFIG)
+        if xml2_version and xslt_version:
+            return xml2_version, xslt_version
+
+    # One or both build dependencies not found.
```

Comment:
This used to be the case for the Windows build with pre-built downloaded libraries. Failing hard here breaks this.
Either it should continue here, or we also need to integrate that part of the build here. I think not failing seems like the easiest way to keep it working for now.

Reply:
I think failing is useful, otherwise the build just fails later on, since the headers are missing.

I'd like to keep the hard failure Linux platforms, but allow Windows builds to keep going.

Reply:
Okay, I've pushed a slightly different solution to this.